### ISSUE TITLE
ROM segments are non-contiguous with LLVM9

### ISSUE THREAD

**Issue by QuLogic** [tinygo-org/tinygo#594]
When building with LLVM 9, all of the smoke tests except wasm and avr fail with this error:
```
tinygo build -size short -o test.hex -target=pca10040            examples/blinky1
   code    data     bss |   flash     ram
   5340      60     196 |    5400     256
error: ROM segments are non-contiguous: /tmp/tinygo703550982/main
```
There's not much more information, so I don't know if LLVM is stricter or tinygo is outputting something wrong now.

**Comment by aykevl** [comment]
Thank you for the report! This is a problem in TinyGo and should be fixed. It looks like the lld linker was changed significantly.

**Comment by aykevl** [comment]
Turns out lld decided to fill the `.stack` section with some arbitrary pattern, and thereby marking it as `PROGBITS` instead of `NOBITS`. This should be fixed in #715.

**Comment by aykevl** [comment]
The fix for this has been merged in the dev branch.
